src/interpolate.jl

```diff
@@ -40,6 +40,113 @@ function cell2node(cell_centered_data::AbstractArray{Float64})
 end
 
 
+# Convert 3d unstructured data to 2d slice.
+# Additional to the new unstructured data updated coordinates, levels and
+# center coordinates are returned.
+function unstructured_2d_to_3d(unstructured_data::AbstractArray{Float64},
+                               coordinates::AbstractArray{Float64},
+                               levels::AbstractArray{Int}, length_level_0::Float64,
+                               center_level_0::AbstractArray{Float64},
+                               slice_axis, slice_axis_intersect)
+
+  dimensions = Dict(
+    :x => 1,
+    :y => 2,
+    :z => 3
+  )
+  if !haskey(dimensions, slice_axis)
+    supported_dims = keys(dimensions)
+    error("illegal dimension '$slice_axis', supported dimensions are $supported_dims")
+  end
+  slice_axis_dimension = dimensions[slice_axis]
+  other_dimensions = [1, 2, 3][1:end .!= slice_axis_dimension]
```

Comment:
I'd suggest to make this code more explicit (and, I'd argue, easier to read) and avoid the temporary data structures (but this is really a suggestion):
```suggestion
  if slice_axis === :x
    slice_axis_dimension = 1
    other_dimensions = (2, 3)
  elseif slice_axis === :y
    slice_axis_dimension = 2
    other_dimensions = (1, 3)
  elseif slice_axis === :z
    slice_axis_dimension = 3
    other_dimensions = (1, 2)
  else
    error("illegal dimension '$slice_axis', supported dimensions are :x, :y, and :z")
  end
```

Reply:
I agree. Can you explain why you are using `===` in this case instead of `==`?

Reply:
From the Julia docs:
>  ===(x,y) -> Bool
>
>  Determine whether x and y are identical, in the sense that no program could distinguish them. First the types of x and y are compared. If those are identical, mutable objects are compared by address in memory
  and immutable objects (such as numbers) are compared by contents at the bit level. This function is sometimes called "egal". It always returns a Bool value.

It works, e.g., for comparing literals and symbols. I don't have a good heuristic about which one to use in which situation, but usually go for "unless you specifically need `===`, just `==`". However, here I have violated this heuristic, so feel free to change that to `==`.